bitwiseio/sawtooth-sdk-java
# Add UI for submitting a create game transaction.

Depends on #3.

Adds the UI for submitting a create game transaction.

#### To test:
1. In `~/sawtooth-sdk-java/examples/xo_android_client/` run `docker-compose up`. This will start the sawtooth network, including the validator, xo-tp, sawtooth-shell, sawtooth-rest-api.

1. Run the App on Android Studio:
   - If you are using a emulator to test the app you don't have to do anything special
   - If you are using a phone to test the app you, on the file XOState line 31, you have to change the URL for the rest api. The new url should be `<your-computer-api-address>:9708`. I was not actually able to test running the app on a phone, as my computer for some reason is not finding my phone when connected via USB.

2. When the app opens click on "+" button.
     <img src="https://user-images.githubusercontent.com/14094978/48726108-f84a4f80-ebf3-11e8-8b74-bec4f3c52deb.png" width="150">

3. Enter a game name and press the send button
    <img src="https://user-images.githubusercontent.com/14094978/48726224-5119e800-ebf4-11e8-8289-33588ea9c63b.png" width="150">

4. A message showing the batch status should appear on the screen shortly, like shown bellow:
    <img src="https://user-images.githubusercontent.com/14094978/48726450-00ef5580-ebf5-11e8-9aba-182e777af0be.png" width="150">

Right now, if the transaction fails all the user will see is "Batch status: INVALID". That can be improved in the future. 

You can also run `docker exec -it sawtooth-shell bash` and `xo list --url http://rest-api:9708` to log into the sawtooth-shell and use the xo client to confirm that the game was created
